### PR TITLE
[GISel][RISCV] Anyextend before copying f16 -> i32/i64

### PR DESCRIPTION
**Pull request by dtcxzyw**
Fixes type check failure in https://github.com/llvm/llvm-project/pull/94110#issuecomment-2158417937

### CODE REVIEW

File: llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp [hunk 1]
```diff
@@ -102,9 +102,16 @@ struct RISCVOutgoingValueHandler : public CallLowering::OutgoingValueHandler {
...
+    // If we're passing a smaller fp value into a larger integer register,
+    // anyextend before copying.
+    if ((VA.getLocVT() == MVT::i64 && VA.getValVT() == MVT::f32) ||
+        ((VA.getLocVT() == MVT::i32 || VA.getLocVT() == MVT::i64) &&
+         VA.getValVT() == MVT::f16))
+      ValVReg =
+          MIRBuilder
+              .buildAnyExt(LLT::scalar(VA.getLocVT().getScalarSizeInBits()),
+                           ValVReg)
+              .getReg(0);
```
Comment (arsenm) [inline llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp:114]: Use some temporary variables and braces? This wrapping is horrible

**Review by michaelmaitland — APPROVED** [review]
LGTM

**Review by arsenm — APPROVED** [review]

File: llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp [hunk 2]
```diff
@@ -102,9 +102,14 @@ struct RISCVOutgoingValueHandler : public CallLowering::OutgoingValueHandler {
...
                         const CCValAssign &VA) override {
-    // If we're passing an f32 value into an i64, anyextend before copying.
-    if (VA.getLocVT() == MVT::i64 && VA.getValVT() == MVT::f32)
-      ValVReg = MIRBuilder.buildAnyExt(LLT::scalar(64), ValVReg).getReg(0);
+    // If we're passing a smaller fp value into a larger integer register,
+    // anyextend before copying.
+    if ((VA.getLocVT() == MVT::i64 && VA.getValVT() == MVT::f32) ||
+        ((VA.getLocVT() == MVT::i32 || VA.getLocVT() == MVT::i64) &&
+         VA.getValVT() == MVT::f16)) {
+      LLT DstTy = LLT::scalar(VA.getLocVT().getScalarSizeInBits());
```
Comment (topperc) [inline llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp:110]: getScalarSizeInBits() -> getSizeInBits(). We already know its a scalar.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**